### ISSUE TITLE
Drum Kit cursor/mouse/click note input

### ISSUE THREAD

**Issue by chriswastyping** [musescore/MuseScore#24055]
### Your idea

As a drummer, I always find that writting drum parts is always a pain. My standard first step in musescore is usually adding a large drum kit from unpitched percussion. Then trying to understand how to input notes, it never feels intuitive.

https://github.com/user-attachments/assets/e8e4fcf8-fb2c-424f-a361-2be50b5d6c42

For some reason in musescore you can't click to input a note, but you can if you press a key i.e B to input a bass drum. It would be better to just be able to click the note into the bar like other instruments in the app. As I understand this is to choose drum sounds? But then the snare drum only offers one note for 3 different types of notation. There's also a default to make sure that the bass drum is seprate from snare and hi-hat - I get the separation but generally everything on voice 1 should work ok.

<img width="1030" alt="Screenshot 2024-08-15 at 17 10 53" src="https://github.com/user-attachments/assets/4d2cb8cf-da99-4d13-878e-ca965dfd3816">

Here's a pretty standard drum kit notation guide: 

https://drummagazine.com/drum-notation-guide/
https://www.drumeo.com/beat/how-to-read-drum-music/

### Problem to be solved

- Overhaul of drum kit input notation
- Condense 4-piec kit, 5-piece kit, large kit to one drum kit
- Use modern standardised drum kit notation

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by zacjansheski** [comment]
There is no true standard for drum kit notation.  Evidenced by how the notations on the provided links don't match eachother.

A better cursor/mouse/click note input system for percussion is a feature we have planned for 4.5

**Comment by MarcSabatella** [comment]
To be clear, though: you *can* click to enter notes onto percussion staves, with no keyboard required.  You just have to use the drum input panel select which note you want to enter first, since often there are multiple notes that can entered onto the same line.  Or you can simply double-click the note in the panel to save the back and forth motion between score and panel.

Room for improvement for sure, so I'm excited to see what they come up with, but it is not the case that you can't enter by  mouse alone.  Of course,e entering by keyboard is far more efficient, so hopefully the limitation to only 7 note input shortcuts is also lifted as part of the redesign.

Anyhow, if you need further help entering percussion more efficiently while waiting for the improvements yet to come, be sure to ask on the forums at musescore.org

**Comment by bobjp** [comment]
True, there is no set standard for drum set notation. For example Sibelius doesn't use the hands and feet style. All drums are in voice one. Cymbals are different voices and noteheads. Notes are intered much like any other instrument. Drum note can be entered quickly and easily with the mouse. As well as shortcuts. 

Sure, in MuseScore drum set notes can be entered by double clicking in the panel. But it is awkward, imprecise, and slow. Un natural as opposed to working in the staff. Which is a large part of the problem with the drum palette.

**Comment by chriswastyping** [comment]
Thanks for getting back folks, just on the notation, I mean more as drummer reading the music, it feels like the palette provided is limited and it just doesn't feel as intuitive compared to sibelius maybe? That being said, I am looking forward to seeing the new input system!

**Comment by MarcSabatella** [comment]
So far the new percussion panel doesn't allow this.  However, the stateless single-click model it implements might have provided a good first step, and the fact that pads can be re-arranged provides a good second step.  I think a pretty simple change would allow for this to actually work as users have long wanted.

Right now in current nightly builds, clicking in the staff does *nothing*, regardless of what pad was most recently clicked - the panel is stateless.  That's great, because it means a) there is no existing behavior we'd be breaking, and b) we don't have to worry about whether the user wanted to enter a note based on the selected pad the selected staff position (it's definitely the staff position).  So, what if we simply entered a note at the current staff position, using the order of the pads to help choose *which* note to enter if there are multiple notes at the same staff position?

I could envision that being a pretty trivial code change, with no additional design work needed because the way the panel works is already compatible with this.  Of course, we *could* improve it by having the "shadow note" actually show the notehead that would be added, and maybe even highlight the corresponding pad.

With this in place, one could easily enter most drum kit parts by mouse without even opening the panel.  Personally, I'd still use keyboard shortcuts, but this seems like a pretty easy win for mouse users.

**Comment by mathesoncalum** [comment]
Resolved in #26000! That PR implements percussion mouse input pretty much as-described above.

On your point about notes at the same staff position @MarcSabatella, we've introduced a popup you can use to cycle noteheads. You can access it by pressing shift while hovering over a percussion staff in note input mode.

Cheers!

**Comment by MarcSabatella** [comment]
Great news!  I had refrained from opening issues about any of this thus far because so much still seemed WIP, but the mention that beta is "any day now" made me think it was more done and worth getting this in.  The changes over the last couple of days are really making a huge difference in my perception, so thanks for all of this!
